### PR TITLE
.includes()-a-regex-alternative.md

### PR DESCRIPTION
**Pull request by SavageWilliam**
Hello All! First pull request so lets see how this goes.
A TIL md file about a .includes() and how it can be used as a RegEx alternative
Check it out, contributions welcome! 
Happy pulling

### CODE REVIEW

**Comment by msachi** [comment]
@SavageWilliam I would suggest adding this to either the string README or the ES6 README. I think the idea of TILs has been more the one of quick tips and snippets. To me this reads more like a (section of a) tutorial. Which is a good thing! :)

**Comment by SavageWilliam** [comment]
@msachi You're right!  I followed the guidelines for 'writing a good TIL'. But now you mention, it seems like either a brief tutorial, or a thorough TIL? Would you suggest I elaborate a little more before adding to READMEs? Also, how do change which string I'm adding it to? Thanks!

**Comment by msachi** [comment]
Cool! Yeah I think the TIL guidelines should be updated, 'max 60 lines' is definitely an overkill... I think it's fine as it is if you were to just add to an existing README, but you'd need to elaborate on it if you wanted to make a new README. Based on previous discussions, it's preferable to improve existing ones, ad we'd like to get to a book-like format eventually, rather than have lots of fragmented topics. I'm not sure I understood your last question, can you rephrase?

**Comment by msachi** [comment]
Also a technical comment: I think you're talking about both String.prototype.include() and Array.prototype.include() here, which are technically different methods. It would be good to make that clear, and keep that in mind if you're adding to String manipulation README (maybe ES6 README is best in that case, so you can talk about both). And I've spotted a few typos :)

**Comment by SavageWilliam** [comment]
Sounds like a good idea! Feel free to lay into any typos you see, or any other improvements you can think of. I've changed the structure a little to separate the different prototypes. Whats the best way to relocate where I am making the pull request for? (Aka to add this to an existing README rather than a new TIL file).

**Comment by SavageWilliam** [comment]
@msachi Feel free to correct any typos you come across! Having spent more time reading some READMEs, I understand your original comment more so. When writing the file, what I had in mind was an illustration as to how the .includes() method can be used  in different ways as an 'alternative' to RegEx. Rather than a tutorial type lesson on the .includes() prototype.  Just as a way to unstick coders who are new to RegEx when faced with a daunting RegEx problem.

**Comment by msachi** [comment]
Typos fixed! :) Yeah, I understand, it's often a good idea to start with a TIL, but I think you've already done too much work and it deserves a README ;) But ultimately up to you. (I'm not aware of a good way to move it there other than copy-pasting, I'm afraid.)

**Comment by msachi** [comment]
Oh and two more things:
- You say "The string contains "world" after position 12" - it should be 'before' right?
- It might be interesting (if you're going to build on this) to think of some examples where includes() is better / more powerful than regex. Because for the example that you provided, I think `'Hello world!'.replace(/[aeiou]/gi,'')` does the job and is shorter to use than your includes() example. Just a thought!

**Comment by SavageWilliam** [comment]
@msachi You are a fine quality control! :D Thanks very much. In general, how many opinions does one sort out before merging a pull request? I suppose it varies, but is there a lower limit?

**Comment by msachi** [comment]
Haha I know, but only when I'm enthusiastic about the topic :) Welcome :) In general getting one person's opinion is enough, except if you're doing something big that will affect everyone (e.g. changing the instructions on how to write READMEs). The idea is that you don't merge your own requests, but get someone else to check through your edits and do the merge.

**Comment by SavageWilliam** [comment]
I'm thinking (if you can't see any further adjustments to the file) we add this to the TIL's and I'll add some info on .includes() to the string-dissection-and-manipulation README file? I feel the angle of the file (aiming to help people who haven't learn some RegEx yet) means it sits better alone. And to fit into that README it needs a more fundamental description and usage? (I'm also conscious to not overrun the READMEs) If you're happy make the merge :)

**Comment by msachi** [comment]
Sure, happy to make the merge. It's way too perfect for a TIL but anyway ;) Maybe you could add it to the ES6 readme instead of the string readme (to keep the arrays), or just make a new readme (there are some readmes already for specific methods)... Up to you.
